src/main/java/seedu/address/model/person/Person.java

```diff
@@ -116,6 +116,29 @@ public Person removeTag(Tag t) {
         return new Person(this.name, this.phone, this.email, newTagSet);
     }
 
+    /**
+     * Adds a tag into a person.
+     * Assumes the person does not contain any tag in the new tag set.
+     */
+    public Person addTag(Set<? extends Tag> tagSet) {
+        Set<Tag> newTagSet = new HashSet<>();
+        newTagSet.addAll(this.getTags());
+        newTagSet.addAll(tagSet);
+        return new Person(this.name, this.phone, this.email, newTagSet);
```

Comment:
We need to find a way to remove the need to add a new person just to edit the tags, perhaps I'll look into it